sensorprocess/sensorprocess.go

```diff
@@ -12,39 +11,85 @@ import (
 	s "github.com/viamrobotics/viam-cartographer/sensors"
 )
 
-// defaultTime is used to check if timestamps have not been set yet.
-var defaultTime = time.Time{}
-
 // Config holds config needed throughout the process of adding a sensor reading to the cartofacade.
 type Config struct {
-	CartoFacade            cartofacade.Interface
-	sensorProcessStartTime time.Time
-	IsOnline               bool
-
-	Lidar            s.TimedLidar
-	currentLidarData *s.TimedLidarReadingResponse
+	CartoFacade cartofacade.Interface
+	IsOnline    bool
 
-	IMU            s.TimedMovementSensor
-	currentIMUData *s.TimedIMUReadingResponse
+	Lidar s.TimedLidar
+	IMU   s.TimedMovementSensor
 
 	Timeout                  time.Duration
 	InternalTimeout          time.Duration
 	Logger                   logging.Logger
 	RunFinalOptimizationFunc func(context.Context, time.Duration) error
-
-	Mutex *sync.Mutex
 }
 
-// Update currentIMUData under a mutex lock.
-func (config *Config) updateMutexProtectedIMUData(data s.TimedIMUReadingResponse) {
-	config.Mutex.Lock()
-	config.currentIMUData = &data
-	config.Mutex.Unlock()
+// StartOfflineSensorProcess starts the process of adding lidar and movement sensor data
+// in a deterministically defined order to cartographer.
+func (config *Config) StartOfflineSensorProcess(ctx context.Context) bool {
```

Comment:
This is the main addition of this PR.

Reply:
We stop the sensor process once the lidar **or** the IMU runs out of data. This was agreed upon in a discussion with @EshaMaharishi .